apps/os/app/routes/org/estate/integrations/mcp-params.tsx

```diff
@@ -24,12 +24,39 @@ export const Route = createFileRoute(
 )({
   component: MCPParams,
   validateSearch: z.object({
-    serverUrl: z.string().catch(""),
-    mode: z.enum(["personal", "company"]).catch("personal"),
-    connectionKey: z.string().catch(""),
-    requiredParams: z.string().catch("[]"),
-    agentDurableObject: z.string().catch(""),
-    integrationSlug: z.string().catch(""),
+    serverUrl: z.string(),
+    mode: z.enum(["personal", "company"]),
+    connectionKey: z.string(),
+    requiredParams: z.preprocess((val) => {
+      if (Array.isArray(val)) return val;
+      if (typeof val === "string" && val) {
+        try {
+          return JSON.parse(val);
+        } catch {
+          try {
+            return JSON.parse(decodeURIComponent(val));
+          } catch {
+            return [];
+          }
+        }
+      }
+      return [];
+    }, z.array(MCPParam)),
+    agentDurableObject: z.preprocess((val) => {
+      if (typeof val === "string" && val) {
+        try {
+          return JSON.parse(val);
+        } catch {
+          try {
+            return JSON.parse(decodeURIComponent(val));
+          } catch {
+            return undefined;
+          }
+        }
+      }
+      return val;
+    }, AgentDurableObjectInfo.optional()),
```

Comment:
??? aren't we the ones creating these params? can't we just encode them in a wway that tanstack accepts without all this craziness?

Reply:
see also this: https://zod.dev/codecs#jsonschema